File: sympy/axiom/equality/discrete/combinatorics/binomial/theorem.py
from sympy.core.symbol import Symbol
from sympy.functions.combinatorial.factorials import binomial
from sympy.core.relational import Equality
from sympy.utility import plausible, cout, Eq

from sympy.utility import Sum
from sympy.axiom.equality.discrete.combinatorics.binomial import Pascal

from sympy.logic.boolalg import plausibles_dict


def apply(x, y, n=None):
    k = Symbol('k', integer=True)
    if n is None:
        n = Symbol('n', integer=True, nonnegative=True)
        return Equality((x + y) ** n, Sum[k:0:n](binomial(n, k) * x ** k * y ** (n - k)), plausible=plausible(), for_clause=n)
    elif n < 0:
        return None
    else:
        return Equality((x + y) ** n, Sum[k:0:n](binomial(n, k) * x ** k * y ** (n - k)), plausible=plausible())


from sympy.utility import check


@check
def prove():
    x = Symbol('x', integer=True)
    y = Symbol('y', integer=True)
    n = Symbol('n', integer=True, nonnegative=True)
    cout << apply(x, y, n)

    cout << Eq[-1].subs(n, n + 1)

    cout << (Eq[-2] * (x + y)).powsimp()

    cout << Eq[-1].subs(Eq[-2])

    cout << Eq[-1].right.as_one_term()

    cout << Eq[-1].right.function.expand()

    cout << Eq[-1].right.function.powsimp()

    (k, *_), *_ = Eq[-1].right.limits
    cout << Eq[-1].right.as_two_terms()

    cout << Eq[-1].right.args[1].subs(k, k - 1)

    cout << Eq[-1].subs(Pascal.apply(n + 1, k))

    cout << Eq[-1].left.expand()
    cout << Eq[-1].left.powsimp(deep=True)

    cout << Eq[-1].left.args[0].simplifier()

    cout << Eq[-1].left.simplifier()

    cout << Eq[0].subs(n, 0).right.doit()


if __name__ == '__main__':
    prove()

    print('plausibles_dict:')
    for index, eq in plausibles_dict(Eq).items():
        print("Eq[%d] : %s" % (index, eq))

# executive
# commander

File: axiom/discrete/combinatorics/binomial/theorem.php
<?php
require_once '..\..\..\..\utility.php';
$i = 0;
$txt[$i++] = "\\(\\left(x + y\\right)^{n} = \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n} {\\binom{n}{k}}\\tag*{Eq[0]}\\)";
$txt[$i++] = "\\(\\left(x + y\\right)^{n + 1} = \\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}}\\tag*{Eq[1]}\\)";
$txt[$i++] = "\\(\\left(x + y\\right)^{n + 1} = \\left(x + y\\right) \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n} {\\binom{n}{k}}\\tag*{Eq[2]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}} = \\left(x + y\\right) \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n} {\\binom{n}{k}}\\tag*{Eq[3]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}} = \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n} \\left(x + y\\right) {\\binom{n}{k}}\\tag*{Eq[4]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}} = \\sum\\limits_{k=0}^{n} \\left(x x^{k} y^{- k} y^{n} {\\binom{n}{k}} + x^{k} y y^{- k} y^{n} {\\binom{n}{k}}\\right)\\tag*{Eq[5]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}} = \\sum\\limits_{k=0}^{n} \\left(x^{k} y^{- k + n + 1} {\\binom{n}{k}} + x^{k + 1} y^{- k + n} {\\binom{n}{k}}\\right)\\tag*{Eq[6]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}} = \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n + 1} {\\binom{n}{k}} + \\sum\\limits_{k=0}^{n} x^{k + 1} y^{- k + n} {\\binom{n}{k}}\\tag*{Eq[7]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n + 1}{k}} = \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n + 1} {\\binom{n}{k}} + \\sum\\limits_{k=1}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n}{k - 1}}\\tag*{Eq[8]}\\)";
$txt[$i++] = "\\(\\sum\\limits_{k=0}^{n + 1} x^{k} y^{- k + n + 1} \\left({\\binom{n}{k}} + {\\binom{n}{k - 1}}\\right) = \\sum\\limits_{k=0}^{n} x^{k} y^{- k + n + 1} {\\binom{n}{k}} + \\sum\\limits_{k=1}^{n + 1} x^{k} y^{- k + n + 1} {\\binom{n}{k - 1}}\\tag*{Eq[9]}\\)";
$txt[$i++] = "\\(\\text{True}\\)";
$txt[$i++] = "\\(\\text{True}\\)";
render(__FILE__, $txt);
?>        

